docs/cli/ilab-rag-retrieval.md

```diff
@@ -0,0 +1,264 @@
+# Ingesting and Utilizing RAG Artifacts Design Proposal
+
+**Author**: Daniele Martinoli
+
+**Version**: 0.1
+
+## 1. Introduction
+Purpose of this document is to propose `ilab` CLI changes to extend the InstructLab flow to generate and utilize RAG artifacts
+in the customer workflows. The proposal includes new commands to ingest a local RAG from pre-processed customer documentation, that we assume to be generated from docling output, and updates to the existing commands to utilize these RAG artifacts.
+
+## 2. Use Case
+* Accelerate POCs of fine-tuning LLMs + RAG.
+* Generate artifacts that data scientists can use to build custom solutions in their development environment:
+  * Jupyter Notebooks & Python Scripts
+  * Podman Desktop
+  * Red Hat OpenShift AI
+  * Internal Red Hat CI systems for products or services (e.g., Lightspeed products)
+
+## 3. Proposed Commands
+**Note**: In the context of version 1.4, currently under development, no changes to the command-line interface should be
+allowed. Therefore, we also propose alternative approaches to run the same RAG pipelines using existing `ilab` commands or
+other provided tools.
+
+### 3.1 RAG Ingestion Pipeline Command
+The proposal is to add a `rag` subgroup under the `data` group, with an `ingest` command, like:
+```
+ilab data rag ingest /path/to/docs/folder
+```
+
+#### Command Purpose
+This command processes embeddings generated from documents located in the */path/to/docs/folder* folder and stores them in a vector database. These embeddings are intended to be used as augmented context in a Retrieval-Augmented Generation (RAG) chat pipeline.
+
+#### Assumptions
+The documents must be in JSON format and pre-processed using the `docling` tool.
+
+**Note**: The expected JSON schema is the same that we have when we generate knowledge data with `ilab data generate` and we
+add a reference document to the `qna.yaml` document(s).
+
+#### Supported Databases
+The command supports multiple vector database types. By default, it uses a local `MilvusLite` instance stored at `./rag-output.db`.
+
+#### Usage
+The generated embeddings can later be retrieved to enrich the context for RAG-based chat pipelines.
+
+#### Running the pipeline with v1.4
+Currently, there is no `ilab data rag` command available to execute the RAG ingestion pipeline. As a result, it is not possible
+to address the limitations of version 1.4 using the configuration options detailed below. 
+Alternatively, we propose providing a Jupyter notebook or a standalone script to run the pipeline with default settings.
+
+The script should guide users in overriding default options and generating the RAG artifacts in the configured vector database instance. By default, support for MilvusLite will be included.
+
+> **ℹ️ CLI-based alternative:** This is an alternative to run both the document transformation and the document ingestion in 
```

Comment:
added one option to overcome today's limitations and run the docs and ingestion pipelines with no CLI changes